catalog/core/catalog-core-migratable/src/test/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImplTest.java

```diff
@@ -0,0 +1,429 @@
+                config);
+
+        MigrationMetadata metadata = migratable.export(EXPORT_PATH);
+
+        verify(taskManager).exportSetup();
+        verify(taskManager).exportMetacardQuery(result1, 1);
+        // TODO - Need to add ArgumentCaptor in when() and assert that query.getStartIndex() == 1
+        verify(taskManager).exportFinish();
+        verifyNoMoreInteractions(taskManager);
+
+        assertThat(metadata.getMigrationWarnings(), is(empty()));
+    }
+
+    @Test
+    public void exportWhenResultSetLargerThanExportPageSize() throws Exception {
+        List<Result> result1 = Arrays.asList(mock(Result.class), mock(Result.class));
+        List<Result> result2 = Collections.singletonList(mock(Result.class));
+        setupProviderResponses(result1, result2);
+
+        CatalogMigratableImpl migratable = new CatalogMigratableImplUnderTest(DESCRIPTION,
+                provider,
+                filterBuilder,
+                fileWriter,
+                config);
+
+        MigrationMetadata metadata = migratable.export(EXPORT_PATH);
+
+        verify(taskManager).exportSetup();
+        verify(taskManager).exportMetacardQuery(result1, 1);
+        verify(query).setStartIndex(config.getExportQueryPageSize() + 1);
+        verify(taskManager).exportMetacardQuery(result2, 2);
+        verify(taskManager).exportFinish();
+
+        assertThat(metadata.getMigrationWarnings(), is(empty()));
+    }
+
+    @Test(expected = ExportMigrationException.class)
+    public void exportStopsWhenExportMetacardFails() throws Exception {
+        List<Result> result1 = Arrays.asList(mock(Result.class), mock(Result.class));
+        List<Result> result2 = Collections.singletonList(mock(Result.class));
+        setupProviderResponses(result1, result2);
+
+        doThrow(new RuntimeException()).when(taskManager)
+                .exportMetacardQuery(result1, 1);
+
+        CatalogMigratableImpl migratable = new CatalogMigratableImplUnderTest(DESCRIPTION,
+                provider,
+                filterBuilder,
+                fileWriter,
+                config);
+
+        try {
+            migratable.export(EXPORT_PATH);
+        } finally {
+            verify(taskManager).exportSetup();
+            verify(taskManager).exportMetacardQuery(result1, 1);
+            verify(taskManager).exportFinish();
+            verifyNoMoreInteractions(taskManager);
+        }
+    }
+
+    @Test(expected = ExportMigrationException.class)
+    public void exportWhenProviderQueryFails() throws UnsupportedQueryException {
+        when(provider.query(any())).thenThrow(new UnsupportedQueryException(""));
+
+        CatalogMigratableImpl migratable = new CatalogMigratableImpl(DESCRIPTION,
+                provider,
+                filterBuilder,
+                fileWriter,
+                config);
+
+        migratable.export(EXPORT_PATH);
+    }
+
+    // TODO - Don't think this is possible right now. Delete or re-enable if it becomes possible.
```

Comment:
Assuming we keep the `checkForFailure()` call in `exportMetacardQuery()`, we should remove this TODO and the `@Ignore` and make sure this test passes.